src/helpers/ides.js

```diff
@@ -157,23 +157,35 @@ module.exports = {
   getVisualStudioInfo: () => {
     utils.log('trace', 'getVisualStudioInfo');
     if (utils.isWindows) {
-      return utils
-        .run(
-          `"${process.env['ProgramFiles(x86)']}/Microsoft Visual Studio/Installer/vswhere.exe" -format json -prerelease`
-        )
-        .then(jsonText =>
-          JSON.parse(jsonText).map(vsInstance => {
-            return { Version: vsInstance.installationVersion, DisplayName: vsInstance.displayName };
-          })
-        )
-        .then(x =>
-          utils.determineFound(
-            'Visual Studio',
-            x.map(v => `${v.Version} (${v.DisplayName})`)
+      try {
```

Comment:
Both try catch can be replaced by Promise.catch ?

Reply:
@gengjiawen interesting - I tried using `.catch()` but I couldn't get the right behavior (i.e. it would go unhandled), but maybe I wasn't doing it right... Can you describe how this should work?

Reply:
I think I figured it out